cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/CoreSchedulerService.java

```diff
@@ -111,21 +121,77 @@ public void addSchedules(final Iterable<? extends ProgramSchedule> schedules) th
       @Override
       public Void run(ProgramScheduleStoreDataset store) throws AlreadyExistsException {
         store.addSchedules(schedules);
+        for (ProgramSchedule schedule : schedules) {
+          try {
+            scheduler.addProgramSchedule(schedule);
+          } catch (SchedulerException e) {
+            // TODO: temporarily catch the SchedulerException and throw AlreadyExistsException.
```

Comment:
yes, it will prevent the program schedules from being added, but it will not prevent the oild schedules from being added. They will be left behind. So this will fail, meybe user fixes his schedules and tries again. Then it will fail again because some schedules already exist in Quartz.